ui/components/app/assets/asset-list/asset-list-control-bar/asset-list-control-bar.tsx

```diff
@@ -28,21 +31,40 @@ import {
   ENVIRONMENT_TYPE_POPUP,
 } from '../../../../../../shared/constants/app';
 import NetworkFilter from '../network-filter';
+import { setTokenNetworkFilter } from '../../../../../store/actions';
+import Tooltip from '../../../../ui/tooltip';
 
 type AssetListControlBarProps = {
   showTokensLinks?: boolean;
 };
 
 const AssetListControlBar = ({ showTokensLinks }: AssetListControlBarProps) => {
   const t = useI18nContext();
+  const dispatch = useDispatch();
   const popoverRef = useRef<HTMLDivElement>(null);
   const currentNetwork = useSelector(getCurrentNetwork);
+  const allNetworks = useSelector(getNetworkConfigurationsByChainId);
   const { tokenNetworkFilter } = useSelector(getPreferences);
   const [isTokenSortPopoverOpen, setIsTokenSortPopoverOpen] = useState(false);
   const [isNetworkFilterPopoverOpen, setIsNetworkFilterPopoverOpen] =
     useState(false);
 
-  const allNetworksFilterShown = Object.keys(tokenNetworkFilter ?? {}).length;
+  const allOpts: Record<string, boolean> = {};
+  Object.keys(allNetworks).forEach((chainId) => {
+    allOpts[chainId] = true;
+  });
+
+  const allNetworksFilterShown =
+    Object.keys(tokenNetworkFilter).length !== Object.keys(allOpts).length;
+
+  // TODO: This useEffect should be a migration
+  // We need to set the default filter for all users to be all included networks, rather than defaulting to empty object
+  // This effect is to unblock and derisk in the short-term
```

Comment:
Is this still true?

Reply:
In a perfect world, I think we should have a migration that runs and updates everyone's default network filter to their selected networks, rather than updating them on the fly in the UI.

I'm not sure if it's totally necessary to be feature complete however. I can remove this comment if it's confusing.